File: frontend/src/types/account.ts
export enum AccountType {
  CHECKING = 'checking',
  SAVINGS = 'savings',
  INVESTMENT = 'investment',
  CREDIT_CARD = 'credit_card',
  CASH = 'cash',
}

export interface AccountBase {
  name: string;
  account_type: AccountType;
  initial_balance: number;
  description?: string;
}

export interface AccountCreate extends AccountBase {}

export interface AccountUpdate {
  name?: string;
  account_type?: AccountType;
  description?: string;
  is_active?: boolean;
}

export interface Account extends AccountBase {
  id: string;
  current_balance: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface AccountSummary {
  id: string;
  name: string;
  account_type: AccountType;
  current_balance: number;
  is_active: boolean;
} 

File: backend/app/models/account.py
from sqlalchemy import Column, String, Numeric, Boolean, DateTime, Enum
from sqlalchemy.orm import relationship
import uuid
from datetime import datetime
from enum import Enum as PyEnum
from ..core.database import Base


class AccountType(PyEnum):
    CHECKING = "checking"
    SAVINGS = "savings"
    INVESTMENT = "investment"
    CREDIT_CARD = "credit_card"
    CASH = "cash"


class Account(Base):
    __tablename__ = "accounts"
    
    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()), index=True)
    name = Column(String(100), nullable=False, index=True)
    account_type = Column(Enum(AccountType), nullable=False)
    initial_balance = Column(Numeric(precision=10, scale=2), nullable=False, default=0)
    current_balance = Column(Numeric(precision=10, scale=2), nullable=False, default=0)
    is_active = Column(Boolean, default=True)
    description = Column(String(255), nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    transactions = relationship("Transaction", back_populates="account")
    recurring_transactions = relationship("RecurringTransaction", back_populates="account")
    
    def __repr__(self):
        return f"<Account(name='{self.name}', type='{self.account_type.value}')>"